### ISSUE TITLE
Atom does not detect the correct PATH.

### ISSUE THREAD

**Issue by ryanpcmcquen** [atom/atom#14924]
### Description

Atom is not inheriting the correct `PATH` for my system.

Terminal:
```
$ echo $PATH
/Applications/Zed.app/bin:/Users/ryan.mcquen/.local/bin:/usr/local/php5/bin:/usr/local/php5-5.6.30-20170506-093822/bin:/usr/local/Homebrew/bin:/opt/X11/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/TeX/texbin:/Applications/Zed.app/bin:/Users/ryan.mcquen/.local/bin:/Users/ryan.mcquen/bin:/usr/local/php5/bin:/usr/local/php5-5.6.30-20170506-093822/bin:/usr/local/Homebrew/bin
```
Atom:
```
process.env.PATH
"/usr/bin:/bin:/usr/sbin:/sbin"
```
### Steps to Reproduce

1. Run Atom.
2. Check your `PATH` (`process.env.PATH` in the Dev Tools Console).
3. See the lack of your `PATH`.

**Expected behavior:** Atom would report the same `PATH` as `echo $PATH` in my terminal.

**Actual behavior:** Atom seems to be using the default system `PATH`.

**Reproduces how often:** 100%

### Versions
```
$ atom --version
Atom    : 1.17.0
Electron: 1.3.15
Chrome  : 52.0.2743.82
Node    : 6.5.0
$ apm --version
apm  1.17.0
npm  3.10.5
node 6.9.5 x64
python 2.7.10
git 2.10.1
```

Mac OS 10.11.6.

### Additional Information

I'm using `tmux`.

**Comment by rsese** [comment]
Just to confirm, do you only see this behavior when using `tmux`?

**Comment by ryanpcmcquen** [comment]
@rsese, yes, if I disable `tmux` the correct `PATH` is picked up.

**Comment by rsese** [comment]
Great thanks for confirming - hopefully someone that uses tmux can try to reproduce (@ungb?) but otherwise I can install it and try it soon.

This also sounds kinda familiar so I'll poke around for existing issues in a bit.

**Comment by ryanpcmcquen** [comment]
Thank you @rsese, I did find this but it was closed: https://github.com/electron/electron/issues/550

I also looked in current issues but there are so many I may have missed it. 👍

**Comment by ryanpcmcquen** [comment]
Also, I've upgraded to `1.18.0` and the issue is still present:

```
$ atom --version
Atom    : 1.18.0
Electron: 1.3.15
Chrome  : 52.0.2743.82
Node    : 6.5.0
```

**Comment by ungb** [comment]
I'm not seeing this behavior...  I opened atom in a tmux session and the path is the same on `process.env.PATH` as it is in `echo $PATH` in terminal.  

You're running `echo`$PATH` in the tmux session too right? 
My steps are as followed:

terminal:
1. run `tmux`
2. run `echo $PATH` and grab the output to compare with atom.
3. run `atom .`

Atom:
1.open dev tools and run `process.env.PATH` in console


Expected result: Compare the contents in atom  and in terminal, should be the same. 


For me it's the same.  Let me know if I'm missing a step @ryanpcmcquen!

**Comment by ryanpcmcquen** [comment]
Have you tried not running Atom from the terminal? I think that is the difference here.

**Comment by ungb** [comment]
Ah I do see it's different if you open atom outside of terminal.  I'll add this to our list of issues to triage and let you know what we find!  

Sorry I read that you were using tmux and thought that was related.

**Comment by ryanpcmcquen** [comment]
Well, strangely enough if I disable tmux and run Atom from the GUI, the `PATH` is picked up correctly.

**Comment by ungb** [comment]
I've added this to our project board for further prioritization, thanks for the info @ryanpcmcquen.

Will keep updates here.

**Comment by fedario** [comment]
I also had this problem and figured it out after a considerable amount of time.In my case the issue was in the way the PATH got exported.

**TLDR: Atom was launched by a non-interactive shell and my PATH additions were only exported to interactive shells.**

My distribution configured .bashrc like this:

```
# If not running interactively, don't do anything
case $- in
    *i*) ;;
      *) return;;
esac

export PATH="..."      # <--- not available to non-interactive shells
````

That means that everything after the case block is ignored if the shell is not interactive. And this was the way atom was run when not started from the terminal. When I exported the PATH before the case block even non-interactive shells would have access to my personal PATH exports.
So, this made the complete PATH available in atom:

```
export PATH="..."      # <--- available to non-interactive shells

# If not running interactively, don't do anything
case $- in
    *i*) ;;
      *) return;;
esac
````

I hope this helps someone who is struggling with this.

**Comment by CarlosEvo** [comment]
Using this [init script](https://github.com/ioquatix/script-runner/blob/master/examples/environment.coffee) seems to solve the issue with tmux

**Comment by ryanpcmcquen** [comment]
@ungb, any progress here?

**Comment by Rua** [comment]
I just ran into this issue. Atom isn't using the path and as a result, cannot find binaries it needs.

**Comment by MLModel** [comment]
Newbie (sort of). Is there any simple way to set the path on a Mac, run as an application (not from the command line)? I have attempted to use atom-runner and a few other things (like executing Mac JXA scripts). For some, the problem was solved by adding paths to my .bashrc. But atom-runner doesn't seem to get the PATH variable from there. I have mac-fix-path installed, which might have helped something when I first got started, but I don't remember what. It seems to me I should be able to specify PATH in config.cson, if not globally at least for each package.

**Comment by aseyffert** [comment]
I don't know what a general solution to this issue would look like (different for each case, I'd wager), but I think an important note to make here is: The `bash` shell's config files are not the system's config files.

In my case I solved the issue by making the relevant additions to my path in `~/.profile` instead of in `~/.bashrc`. (I suspect that) the latter file is only loaded when I open up a `bash` shell (via Konsole, in my case), and not when Atom is launched via the KDE menus (I'm using KDE on Slackware; if Atom is launched from within a `bash` shell, then it inherits the PATH set in `.bashrc`).

I don't know what the analogue to `~/.profile` is on all the various systems, but I'd start by figuring that out when this issue pops up next for me.

**Comment by wjuniorw** [comment]
I had fixed this issue by change Exec propertie in ` /usr/share/applications/atom.desktop `

```
# Exec=env ATOM_DISABLE_SHELLING_OUT_FOR_ENVIRONMENT=false /usr/bin/atom %F
Exec=source $HOME/.bashrc && /usr/bin/atom %F
```
